src/utils/socket_operations.py

```diff
@@ -147,6 +144,37 @@ def send_bytes(socket, receiver_identity, bytes_object, log=None):
     socket.send_multipart(frames)
 
 
-send_obj = send_exp = send_bytes
+def recv_pyobj(socket, sender_identity, log, expected_type=None):
+    """Un-packs a pickled object received from recv_bytes. Can be used
+    to check if the received object is of the expected type.
 
-recv_obj = recv_exp = recv_bytes
+    Args:
+        socket:
+        expected_type:
+        log:
+    """
+    bytes_packet = recv_bytes(socket, sender_identity, log)
+    message = pickle.loads(bytes_packet)
+    if expected_type:
```

Comment:
I prefer `if expected_type is not None:`, since it is more clear what you're checking. `expected_type` is not a boolean